libraries/system/storage/database/personas/upsert.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * This file is part of Bott.
+ *
+ * This project is dual-licensed:
+ * - Non-commercial use: AGPLv3 (see LICENSE file for full text).
+ * - Commercial use: Proprietary License (contact D@nielLaCos.se for details).
+ *
+ * Copyright (C) 2025 DanielLaCos.se
+ */
+
+import type { BottPersona } from "@bott/model";
+
+import { commit } from "../commit.ts";
+import { sql } from "../sql.ts";
+
+export const upsertPersona = (persona: BottPersona): void => {
+  const result = commit(
+    sql`
+      insert into personas (id, user_id, display_name, handle, space_id)
+      values (
+        ${persona.id},
+        ${persona.user?.id ?? null},
```

Comment:
we should also add the user/space if they don't exist yet.

The user's canonical name can just be the one provided on the persona for now.